ClientApp/src/app/login/login.component.ts

```diff
@@ -1,69 +1,76 @@
-import { Component, OnInit } from '@angular/core';
-import { Form, FormGroup, FormBuilder, Validators } from '@angular/forms';
+import {
+  Component,
+  OnInit
+} from '@angular/core';
+import {
+  FormGroup,
+  FormBuilder,
+  Validators
+} from '@angular/forms';
 import { Router } from '@angular/router';
+
+import { MessageService } from 'primeng/api';
+
 import { AuthService } from '../server-api/auth.service';
 import { UserService } from '../server-api/user.service';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
-  styleUrls: ['./login.component.css']
+  styleUrls: ['./login.component.scss'],
+  providers: [MessageService],
 })
 export class LoginComponent implements OnInit {
-
-  // for showing and hiding the password
-  type: string = "password" // used to replace text with password in login.component.html
-  isText: boolean = false
-  showIcon: string = "fa-eye-slash"
-
   //formGroup declared in html
-  loginForm!: FormGroup
+  loginForm!: FormGroup;
+
+  loginButtonLoading: boolean = false;
   
-  constructor(private formbuilder: FormBuilder, private  auth : AuthService, private router: Router, private user: UserService) { }
+  constructor(
+    private formbuilder: FormBuilder,
+    private auth: AuthService,
+    private router: Router,
+    private user: UserService,
+    private messageService: MessageService,
+  ) { }
 
   ngOnInit(): void {
+    this.initLoginForm();
+  }
 
+  initLoginForm() {
     // from html we set the formControl name to corresponding names (username, password)
     this.loginForm = this.formbuilder.group({
       username: ['', Validators.required],
       password: ['', Validators.required]
-    })
-  }
-
-  showOrHidePassword(){
```

Comment:
Is there functionality added to show and hide the password elsewhere?

Reply:
Yup showing/hiding the password is built into primeng's <p-password> component